app/mattermost.js

```diff
@@ -513,14 +516,19 @@ const fromPushNotification = Platform.OS === 'android' && Initialization.replyFr
 if (startedSharedExtension || fromPushNotification) {
     // Hold on launching Entry screen
     app.setAppStarted(true);
-
-    // Listen for when the user opens the app
-    new NativeEventsReceiver().appLaunched(() => {
-        app.setAppStarted(false);
-        launchEntry();
-    });
 }
 
 if (!app.appStarted) {
     launchEntry();
 }
+
+new NativeEventsReceiver().appLaunched(() => {
```

Comment:
Why is this changing as well? This is going to conflict with the react-native-navigiation upgrade

Reply:
I know this will conflict only with the name of the functions that were changed but the concept is the same.

on Android when you have hit the hardware back button the activity gets destroyed, the same is happening when you switch to a split view. Although the activity is recreated the react context already exits thus it executes app launched so we can re-create the views in the app, which is done with `startSingleApp` and we want this to be available at all times instead of only when the app was started by a push notification or the extension